src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/CmsSignature.ECDsa.cs

```diff
@@ -65,8 +65,9 @@ protected override bool VerifyKeyType(AsymmetricAlgorithm key)
                     return false;
                 }
 
-                // 2 * KeySizeBytes => 2 * KeySizeBits / 8 => KeySizeBits / 4
-                int bufSize = key.KeySize / 4;
+                // fieldSize = ceil(KeySizeBits / 8);
+                int fieldSize = ((key.KeySize - 1) / 8) + 1;
```

Comment:
Would `bufSize = (key.KeySize + 3) / 4` work to round up to a multiple of 4?

Reply:
Not quite, since the rounding has to apply per-field.

For secp521r1:

* 521 / 4 => 130
* (521 + 3) / 4 => 524 / 4 => 131
* ((521 + 7) / 8) * 2 => (528 / 8) * 2 => 66 * 2 => 132
* (((521 - 1) / 8) + 1) * 2 => ((520 / 8) + 1) * 2 => (65 + 1) * 2 => 66 * 2 => 132


`int fieldSize = checked((key.KeySize + 7) / 8);` is our normal "how many bytes does this take" formula; but I don't see a place (other than KeySize == 0) where subtract/divide/add gives a different answer than add/divide

Reply:
I can add tests as well, if desired.